test/v1/controllers/AmendOtherReliefsControllerSpec.scala

```diff
@@ -0,0 +1,231 @@
+      |        "reliefClaimed": 763.00
+      |      }
+      |  ]
+      |}""".stripMargin)
+
+  private val requestBody = AmendOtherReliefsBody(
+    Some(NonDeductableLoanInterest(
+    Some("myref"),
+    763.00)),
+    Some(PayrollGiving(
+      Some("myref"),
+      154.00)),
+    Some(QualifyingDistributionRedemptionOfSharesAndSecurities(
+      Some("myref"),
+      222.22)),
+    Some(Seq(MaintenancePayments(
+      "myref",
+      Some("Hilda"),
+      Some("2000-01-01"),
+      Some(222.22)))),
+    Some(Seq(PostCessationTradeReliefAndCertainOtherLosses(
+      "myref",
+      Some("ACME Inc"),
+      Some("2019-08-10"),
+      Some("Widgets Manufacturer"),
+      Some("AB12412/A12"),
+      Some(222.22)))),
+    Some(AnnualPaymentsMade(
+      Some("myref"),
+      763.00)),
+    Some(Seq(QualifyingLoanInterestPayments(
+      "myref",
+      Some("Maurice"),
+      763.00)))
+  )
+
+  private val rawData = AmendOtherReliefsRawData(nino, taxYear, requestJson)
+  private val requestData = AmendOtherReliefsRequest(Nino(nino), taxYear, requestBody)
+
+
+  "handleRequest" should {
+    "return OK" when {
+      "the request received is valid" in new Test {
+
+        MockAmendOtherReliefsRequestParser
+          .parseRequest(rawData)
+          .returns(Right(requestData))
+
+        MockAmendOtherReliefsService
+          .amend(requestData)
+          .returns(Future.successful(Right(ResponseWrapper(correlationId, ()))))
+
+        MockHateoasFactory
+          .wrap((), AmendOtherReliefsHateoasData(nino, taxYear))
+          .returns(HateoasWrapper((), Seq(testHateoasLink)))
+
+        val result: Future[Result] = controller.handleRequest(nino, taxYear)(fakePostRequest(requestJson))
+        status(result) shouldBe OK
+        header("X-CorrelationId", result) shouldBe Some(correlationId)
+      }
+    }
+    "return the error as per spec" when {
+      "parser errors occur" should {
+        def errorsFromParserTester(error: MtdError, expectedStatus: Int): Unit = {
+          s"a ${error.code} error is returned from the parser" in new Test {
+
+            MockAmendOtherReliefsRequestParser
+              .parseRequest(rawData)
+              .returns(Left(ErrorWrapper(Some(correlationId), error, None)))
+
+            val result: Future[Result] = controller.handleRequest(nino, taxYear)(fakePostRequest(requestJson))
+
+            status(result) shouldBe expectedStatus
+            contentAsJson(result) shouldBe Json.toJson(error)
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+          }
+        }
+
+        val input = Seq(
+          (BadRequestError, BAD_REQUEST),
+          (NinoFormatError, BAD_REQUEST),
+          (RuleTaxYearRangeInvalidError, BAD_REQUEST),
+          (ValueFormatError.copy(paths = Some(Seq(
+            "vctSubscription/0/amountInvested",
+            "vctSubscription/0/reliefClaimed"))), BAD_REQUEST),
+          (ReliefDateFormatError, BAD_REQUEST)
```

Comment:
Also need tests for

- FORMAT_TAX_YEAR
- RULE_INCORRECT_OR_EMPTY_BODY_SUBMITTED
- FORMAT_CUSTOMER_REF